met-web/src/components/survey/building/SurveyLoader.tsx

```diff
@@ -0,0 +1,28 @@
+import { EmailVerification } from 'models/emailVerification';
+import { Params, defer } from 'react-router-dom';
+import { getEmailVerification } from 'services/emailVerificationService';
+import { getEngagement } from 'services/engagementService';
+import { getSlugByEngagementId } from 'services/engagementSlugService';
+import { getSubmission, getSubmissionByToken } from 'services/submissionService';
+import { getSurvey } from 'services/surveyService';
+import { fetchSurveyReportSettings } from 'services/surveyService/reportSettingsService';
+
+export const SurveyLoader = async ({ params }: { params: Params<string> }) => {
+    const { surveyId, token, language } = params;
+    if (isNaN(Number(surveyId)) && !token) throw new Error('Invalid survey ID');
+    const verification = getEmailVerification(token ?? '').catch(() => null);
+    const survey = surveyId
+        ? getSurvey(Number(surveyId))
+        : verification.then((response) => {
+              if (!response) throw new Error('Invalid token');
+              return getSurvey(response.survey_id);
```

Comment:
Does this return a survey even if the credentials aren't valid?

Reply:
Good point - updated